Receive/src/bms-rx-exec.c

```diff
@@ -0,0 +1,115 @@
+#include "bms-rx-data.h"
+
+
+/************************ Function Definitions **************************/
+void sighandler(int signum) {
+    printf("User stopped reading, i.e, Signum value : %d received", signum);
+    isStopRequestedByUser = true;
+}
+
+float movingAverageValue(float *ptrArrNumbers, float *ptrSum, int paramID, int range, float nextNum)
```

Comment:
It takes time to understand this function. Think of how you would describe this functionality to someone: "Average of last `range` values. Try to make the code talk in that way. Your complexity also will reduce.